scripts/ci/operator.sh

```diff
@@ -0,0 +1,184 @@
+  kubectl -n "$ns" get pods -l name=istio-operator "${args[*]}"
+}
+
+edit_operator() {
+  echo "Edit the Operator deployment"
+  echo
+
+  local ns=${ns:-openshift-operators}
+  echo "Edit istio-operator Deployment with kubectl edit"
+  kubectl -n "$ns" edit "deploy/istio-operator"
+}
+
+logs_operator() {
+  echo "Display the Operator logs"
+  echo
+
+  local ns=${ns:-openshift-operators}
+  kubectl -n "$ns" logs deploy/istio-operator "${args[*]}"
+}
+
+restart_operator() {
+  echo "Restart the operator."
+  echo
+
+  local ns=${ns:-openshift-operators}
+  echo "Restart operator by deleting the Pod"
+  kubectl -n "$ns" delete pod -l name=istio-operator --force --grace-period 0
+}
+
+# we can remove this function in testing 2.4
+wait_operator() {
+  echo "Temporary workaround before operator 2.4 release"
+  echo
+
+  echo -n "Waiting for Jaeger operator deployment to be created..."
+  while ! kubectl get deployment -n openshift-operators -o name | grep jaeger >& /dev/null ; do echo -n '.'; sleep 1; done
+  echo "done."
+  jaeger_deployment="$(kubectl get deployment -n openshift-operators -o name | grep jaeger)"
+
+  echo -n "Waiting for Kiali operator deployment to be created..."
+  while ! kubectl get deployment -n openshift-operators -o name | grep kiali >& /dev/null ; do echo -n '.'; sleep 1; done
+  echo "done."
+  kiali_deployment="$(kubectl get deployment -n openshift-operators -o name | grep kiali)"
+
+  echo -n "Waiting for Service Mesh operator deployment to be created..."
+  while ! kubectl get deployment -n openshift-operators -o name | grep istio >& /dev/null ; do echo -n '.'; sleep 1; done
+  echo "done."
+  servicemesh_deployment="$(kubectl get deployment -n openshift-operators -o name | grep istio)"
+
+  echo "Waiting for CRDs to be established."
+  for crd in servicemeshcontrolplanes.maistra.io servicemeshmemberrolls.maistra.io jaegers.jaegertracing.io
+  do
+    echo -n "Waiting for CRD [${crd}]..."
+    while ! kubectl get crd "${crd}" >& /dev/null ; do echo -n '.'; sleep 1; done
+    kubectl wait --for condition=established crd/"${crd}"
+  done
+
+  echo "Waiting for operator deployments to start..."
+  for op in "${servicemesh_deployment}" "${jaeger_deployment}"
+  do
+    echo -n "Waiting for ${op} to be ready..."
+    readyReplicas="0"
+    while [ "$?" != "0" -o "$readyReplicas" == "0" ]
+    do
+      sleep 1
+      echo -n '.'
+      readyReplicas="$(kubectl get ${op} -n openshift-operators -o jsonpath='{.status.readyReplicas}' 2> /dev/null)"
+    done
+    echo "done."
+  done
+
+  echo "Wait for the servicemesh validating webhook to be created."
+  while [ "$(kubectl get validatingwebhookconfigurations -o name | grep servicemesh)" == "" ]; do echo -n '.'; sleep 5; done
+  echo "done."
+
+  echo "Wait for the servicemesh mutating webhook to be created."
+  while [ "$(kubectl get mutatingwebhookconfigurations -o name | grep servicemesh)" == "" ]; do echo -n '.'; sleep 5; done
+  echo "done."
```

Comment:
This currently runs against the 2.3.x operator, which doesn't report its readiness properly, so we need to ensure that these webhooks are actually ready. Waiting only for the webhookconfigurations to be created is not enough.

We should add an additional check that actually checks if the webhooks are responding.